src/shared/getCustomJsFunctions.ts

```diff
@@ -0,0 +1,22 @@
+import { getBrowserWindow } from 'cbioportal-frontend-commons';
+import { IOncoprintProps } from 'shared/components/oncoprint/Oncoprint';
+import { DataUnderMouse } from 'shared/components/oncoprint/makeGeneticTrackTooltip';
+
+export type CustomJsFunctions = {
+    createCustomOncoprintTooltip?: (
+        oncoprintProps: IOncoprintProps,
+        dataUnderMouse: DataUnderMouse
+    ) => string;
+};
+
+export default function getCustomJsFunctions(): CustomJsFunctions {
+    if (!getBrowserWindow().customJsFunctions) {
+        getBrowserWindow().customJsFunctions = {};
+    }
+    return getBrowserWindow().customJsFunctions;
+}
+
+// expose functions to dynamically loaded js:
+const win = window as any;
```

Comment:
why do they need these?

Reply:
So the custom js file can append the function to the window object, like this:
```js
getCustomJsFunctions().createCustomOncoprintTooltip = (oncoprintProps, dataUnderMouse) => {
    // create tooltip...
    return $tooltip;
}
```